aten/src/ATen/native/Linear.cpp

```diff
@@ -15,6 +15,14 @@ namespace at { namespace native {
 
 Tensor linear(const Tensor& input, const Tensor& weight, const Tensor& bias) {
   if (input.is_mkldnn()) {
+    TORCH_CHECK(at::globalContext().userEnabledMkldnn(),
+        "MKLDNN linear module can't run when mkldnn disabled");
+    TORCH_CHECK(input.options().type_equal(weight.options()),
+        "Input type (", input.toString(), ") and weight type (", weight.toString(),
+        ") should be the same");
+    TORCH_CHECK(!bias.defined() || (input.options().type_equal(bias.options())),
+        "Input type (", input.toString(), ") and bias type (", bias.toString(),
+        ") should be the same");
     return at::mkldnn_linear(input, weight, bias);
```

Comment:
For linear:
```
x = torch.randn(56, 56, dtype=torch.float)
linear = torch.nn.Linear(56, 56 bias=True)
mkldnn_linear = mkldnn_utils.to_mkldnn(copy.deepcopy(linear))
with torch.backends.mkldnn.flags(enabled=True):
    y_mkldnn = mkldnn_linear(x)
```
1. torch.backends.mkldnn.flags(enabled=True), input and weight are MKLDNN tensor, it can works.
2. torch.backends.mkldnn.flags(enabled=True), input is dense tensor,  weight are MKLDNN tensor, it can works, because we will first transfer input to MLDNN tensor in https://github.com/pytorch/pytorch/blob/bdc00196d1b63a339a5b9b6aaec44cefe069a6ea/torch/utils/mkldnn.py#L30.
3.  torch.backends.mkldnn.flags(enabled=False), input and weight are MKLDNN tensor, it can report error: **MKLDNN linear module can't run when mkldnn disabled**
4. torch.backends.mkldnn.flags(enabled=False), input is dense tensor, weight are MKLDNN tensor, it can also report error: **MKLDNN linear module can't run when mkldnn disabled**.